src/components/CheckLanguageAvailability.js

```diff
@@ -0,0 +1,111 @@
+// @flow
+
+import React, { useState } from "react";
+import { useSelector } from "react-redux";
+import { Trans, useTranslation } from "react-i18next";
+import { View } from "react-native";
+import Locale from "react-native-locale";
+import { useTheme } from "@react-navigation/native";
+import i18next from "i18next";
+
+import BottomModal from "./BottomModal";
+import Button from "./Button";
+import Circle from "./Circle";
+import ModalBottomAction from "./ModalBottomAction";
+import LanguageIcon from "../icons/Language";
+import { languageSelector } from "../reducers/settings";
+import { pushedLanguages } from "../languages";
+import { useLanguageAvailableChecked } from "../context/Locale";
+import { Track } from "../analytics";
+
+export default function CheckLanguageAvailability() {
+  const { t } = useTranslation();
+  const { colors } = useTheme();
+  const [modalOpened, setModalOpened] = useState(true);
+  const { localeIdentifier } = Locale.constants();
+  const osLanguage = localeIdentifier.split("_")[0];
+  const currAppLanguage = useSelector(languageSelector);
+  const [hasAnswered, answer] = useLanguageAvailableChecked();
+
+  const toShow =
+    modalOpened &&
+    !hasAnswered &&
+    currAppLanguage !== osLanguage &&
+    pushedLanguages.includes(osLanguage);
+
+  if (!toShow) {
+    return null;
+  }
+
+  const onRequestClose = () => {
```

Comment:
maybe use Callback here to not re-render these callbacks :thinking: 
the show or not show doesn't impact it so you can still do hooks and have a condition afterwards

Reply:
done https://github.com/LedgerHQ/ledger-live-mobile/pull/1926/commits/2b2ebb9aa33a2836d9e2181f28084af19b01debd